### ISSUE TITLE
Consider removing JANSI native dependency for JDK 23+

### ISSUE THREAD

**Issue by Karm** [quarkusio/quarkus#43912]
### Description

[Optional](https://bugs.openjdk.org/browse/JDK-8299875) in JDK 21 (`-Djdk.console=jdk.internal.le`) and *On* by default in contemporary JDK (tried with JDK 23) is the capability to control terminals, either by ESC byte sequences or in case of Windows cmd, by talking to native Kernel32 Windows API. You can clear lines, set colors, background/foreground etc.
It is achieved by formerly standalone JLine library.

I propose that we might not need this dependency in Quarkus e.g. beginning JDK 23 and we might get rid of it, reducing the complexity of the project. Mind that JANSI pulls native .so libraries to Quarkus and if we could get be without it the better...

### Implementation ideas

Implementation idea is to browse Quarkus code, especially dev tools and devmode and see what we do with the terminal and if it's basic stuff like clear last line, set/reset color, we keep those ESC codes and leave it to Java runtime to either spit those out to terminals that understand (Linux, Cmder on Windows) or to translate to Kernel32 API calls (Windows cmd).

**Comment by gsmet** [comment]
This is very interesting and thanks for opening this issue as we had a lot of issues with Jansi - especially with dev mode class loaders.

If it works, I think I would even be convinced that we try to enable it by default with Java 21 if doable and sufficiently stable.

**Comment by geoand** [comment]
> If it works, I think I would even be convinced that we try to enable it by default with Java 21 if doable and sufficiently stable.

Yeah, when we move to a JDK 21 baseline, it would be nice to have a look and see if we can depend on this
